src/Engines/TNTSearchEngine.php

```diff
@@ -184,19 +222,43 @@ public function map($results, $model)
             return Collection::make();
         }
 
-        $keys   = collect($results['ids'])->values()->all();
-        $fieldsWheres = array_keys($this->builder->wheres);
-        $models = $model->whereIn(
+        $keys = collect($results['ids'])->values()->all();
+
+        $builder = $this->getBuilder($model);
+
+        $models = $builder->whereIn(
             $model->getQualifiedKeyName(), $keys
         )->get()->keyBy($model->getKeyName());
 
         return collect($results['ids'])->map(function ($hit) use ($models) {
-            return $models->has($hit) ? $models[$hit] : null;
-        })->filter(function ($model) use ($fieldsWheres) {
-            return !is_null($model) && array_reduce($fieldsWheres, function ($carry, $item) use($model) {
-                    return $carry && $model[$item] == $this->builder->wheres[$item];
-                }, true);
-        });
+            if (isset($models[$hit])) {
+                return $models[$hit];
+            }
+        })->filter()->values();
+    }
+
+    /**
+     * Return query builder either from given constraints, or as
+     * new query. Add where statements to builder when given.
+     *
+     * @param \Illuminate\Database\Eloquent\Model $model
+     *
+     * @return Builder
+     */
+    public function getBuilder($model)
+    {
+        // get query as given constraint or create a new query
+        $builder = $this->builder->constraints ?? $model->newQuery();
```

Comment:
@michaelklopf could you please change the null coalescing operator to a regular isset(), so that we're backward compatible. currently this breaks the driver on PHP 5.6

Reply:
See https://github.com/teamtnt/laravel-scout-tntsearch-driver/pull/167